src/useBlockheadsList.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from "react";
+import { Blockhead } from "./types";
+import { useEthers } from "@usedapp/core";
+import useBlockheadsContract from "./useBlockheadsContract";
+import { fetchBlockhead } from "./BlockheadsUtil";
+
+export enum LoadingState {
+  FETCHING_BALANCE,
+  GETTING_TOKEN_IDS,
+  DETACHING_PARTS,
+  LOADED,
+}
+
+export default function useBlockheadsList(address?: string) {
```

Comment:
Saw this logic repeated in a few places, so I created this hook that we can reuse elsewhere